pkg/controller/directvolumemigration/rsync.go

```diff
@@ -792,6 +792,79 @@ func (t *Task) getRsyncOptions() []string {
 	return rsyncOpts
 }
 
+type PVCWithSecurityContext struct {
+	name               string
+	fsGroup            *int64
+	supplementalGroups []int64
+	seLinuxOptions     *corev1.SELinuxOptions
+
+	// TODO:
+	// add capabilities for dvm controller to handle case the source
+	// application pods is privileged with the following flags from
+	// PodSecurityContext and Containers' SecurityContext
+	// We need to
+	// 1. go through the pod's volume.
+	// 2. find the container where it is volume mounted.
+	//    i. if the container's fields are non-nil, use that
+	//    ii. if the container's fields are nil, use the pods fields.
+
+	// RunAsUser    *int64
+	// RunAsGroup   *int64
+	// RunAsNonRoot *bool
+}
+
+// Get fsGroup per PVC
+func (t *Task) getfsGroupMapForNamespace() (map[string][]PVCWithSecurityContext, error) {
+	pvcMap := t.getPVCNamespaceMap()
+	pvcSecurityContextMap := map[string][]PVCWithSecurityContext{}
+	for ns, _ := range pvcMap {
+		pvcSecurityContextMap[ns] = []PVCWithSecurityContext{}
+	}
+	for ns, pvcs := range pvcMap {
+		srcClient, err := t.getSourceClient()
+		if err != nil {
+			return nil, err
+		}
+		podList := &corev1.PodList{}
+		err = srcClient.List(context.TODO(), &k8sclient.ListOptions{Namespace: ns}, podList)
+		if err != nil {
+			return nil, err
+		}
+		for _, claimName := range pvcs {
+			for _, p := range podList.Items {
```

Comment:
nit: premature optimization, make a map from the pods list to `map[vol.PersistentVolumeClaim.ClaimName]Pod or to PodSecurityContext and then you can look up in the map instead of doing multiple loops?

Reply:
+1 this optimization is agreed upon especially since we dont have cache enabled for remote clusters. will produce o(n) + o(m) on number of pods(n) and number of pvc(m)